src/variety/interbd.js

```diff
@@ -0,0 +1,699 @@
+//
+
+/* global Set:false */
+
+(function(pidlist, classbase) {
+	if (typeof module === "object" && module.exports) {
+		module.exports = [pidlist, classbase];
+	} else {
+		pzpr.classmgr.makeCustom(pidlist, classbase);
+	}
+})(["interbd"], {
+	//---------------------------------------------------------
+	// マウス入力系
+	MouseEvent: {
+		use: true,
+		inputModes: {
+			edit: ["number", "color", "color-", "clear"],
+			play: ["shade", "unshade"]
+		},
+		mouseinput_auto: function() {
+			if (this.puzzle.playmode) {
+				if (this.mousestart || this.mousemove) {
+					this.inputcell();
+				}
+			} else if (this.puzzle.editmode) {
+				if (this.mousestart) {
+					this.mouseinput_numcolor();
+				}
+			}
+		},
+
+		mouseinput_numcolor: function() {
+			var cell = this.getcell();
+			if (cell.isnull || cell === this.mouseCell) {
+				return;
+			}
+
+			if (cell !== this.cursor.getc()) {
+				this.setcursor(cell);
+			} else {
+				this.setNewNumColor(cell);
+				cell.draw();
+			}
+			this.mouseCell = cell;
+		},
+
+		setNewNumColor: function(cell) {
+			var current = cell.getNumColor();
+			if (this.btn === "left") {
+				if (current === 65) {
+					cell.setNumColor(-1);
+				} else if (current === -1) {
+					cell.setNumColor(0);
+				} else if (current >= 60) {
+					cell.setNumColor((current % 10) + 1);
+				} else {
+					cell.setNumColor(current + 10);
+				}
+			} else if (this.btn === "right") {
+				if (current === -1) {
+					cell.setNumColor(65);
+				} else if (current === 0) {
+					cell.setNumColor(-1);
+				} else if (current < 10) {
+					cell.setNumColor(current + 59);
+				} else {
+					cell.setNumColor(current - 10);
+				}
+			}
+		},
+
+		getNewNumber: function(cell, num) {
+			var color = cell.ques !== 0;
+			var max = cell.getmaxnum(),
+				min = cell.getminnum(),
+				val = -1;
+
+			if (this.btn === "left") {
+				if (num >= max) {
+					val = color ? -2 : -1;
+				} else if (num === -1) {
+					val = -2;
+				} else if (num < min) {
+					val = min;
+				} else {
+					val = num + 1;
+				}
+			} else if (this.btn === "right") {
+				if (num === -1 || num > max || (num === -2 && color)) {
+					val = max;
+				} else if (num <= min) {
+					val = -2;
+				} else if (num === -2) {
+					val = -1;
+				} else {
+					val = num - 1;
+				}
+			}
+			return val;
+		},
+
+		mouseinput_other: function() {
+			if (
+				this.mousestart &&
+				(this.btn === "left" || this.btn === "right") &&
+				(this.inputMode === "color" || this.inputMode === "color-")
+			) {
+				this.mouseinput_color();
+			}
+		},
+
+		mouseinput_color: function() {
+			var cell = this.getcell();
+			if (cell.isnull || cell === this.mouseCell) {
+				return;
+			}
+
+			if (cell !== this.cursor.getc()) {
+				this.setcursor(cell);
+			} else {
+				this.setNewQues(cell);
+				cell.draw();
+			}
+			this.mouseCell = cell;
+		},
+
+		setNewQues: function(cell) {
+			var num = cell.ques,
+				qnum = cell.qnum;
```

Comment:
I know it's not usually done in the code base, but do you think you could add a comment somewhere (top of the file maybe) summarizing the meaning assigned to the various cell properties? I.e., what does qnum, ques, qans etc. mean for this puzzle type Right now it would save me a bit of deciphering, and I think it'd generally be useful for readability of the source code.

Reply:
Also, the input logic here seems a bit subtle, would be nice to have an input test or two.